pallets/collator-assignment/src/lib.rs

```diff
@@ -292,6 +308,23 @@ pub mod pallet {
                 }
             };
 
+            let mut assigned_containers = new_assigned.container_chains.clone();
+            assigned_containers.retain(|_, v| !v.is_empty());
+
+            // On congestion (not enough collators for all paras) prioritized chains need to pay
+            // the minimum tip of the prioritized chains
+            let maybe_tip: Option<BalanceOf<T>> =
+                if assigned_containers.len() < container_chain_ids.len() + parathreads.len() {
```

Comment:
Shouldn't this calculation take into account min collators required per chain? And congestion can happen if sum of min collators required for all chains is greater than assigned containers?

Right now it is set as `collators_per_container` for all chains, so it can simply be `container_chain_ids.len() * collators_per_container + parathreads.len()`

Reply:
I agree here, my impression is that first we try to assign, then if we detect that the assignation does not cover all the demand we sort by tip. So I think it's better to do it as @ParthDesai describes, and assign only once sorted

Reply:
At this point it doesn't matter the min collators required because that was already takin into account in the assignment function (`assign_collators_rotate_all` and `assign_collators_always_keep_old`). So if the amount of assigned chains is less than the demand then it's because the assignment wasn't able to satisfy the minimum demand.
Nevertheless, I have [changed the congestion condition in the latest commit](https://github.com/moondance-labs/tanssi/pull/475/files#diff-8686f26c396cd2f0d9520d4e109f7f2f163dd0635de8cd2c37c84e4b98e53976R248) and the tip check is using that.